app/code/community/Aoe/DbRetry/Resource/Db/Pdo/Mysql/Adapter.php

```diff
@@ -35,6 +35,7 @@ public function query($sql, $bind = [])
                     $try = $maxTries;
                 } catch (Exception $e) {
                     $try++;
+                    Mage::dispatchEvent('aoe_dbretry_exception', ['try' => $try, 'exception' => $e]);
```

Comment:
I think my only input here would be to document the event in the readme and be VERY clear that further DB calls in the event handler are unsupported and dangerous.